internal/chezmoi/sourcestate.go

```diff
@@ -1692,20 +1703,23 @@ func (s *SourceState) getExternalData(
 	var errs []error
 
 	if external.Checksum.Size != 0 {
+		s.warnFunc("%s: warning: insecure size check will be removed, use a secure hash like SHA256 instead\n", externalRelPath)
```

Comment:
`size` should be permitted *in conjunction* with the more secure `sha256` et al  as it would help prevent length extension attacks. (IIRC, `sha384` is resistant.)

Reply:
Thanks for the review. I've update the PR to allow size when specified with a secure hash like SHA256, SHA384, or SHA512.